openg2p_registration/models/openg2p_registration.py

```diff
@@ -581,6 +590,123 @@ def create_beneficiary_from_registration(self):
             "context": context,
         }
 
+    @api.multi
+    def find_duplicates(self):
+        print("Finding Duplicates.......")
+        my_list = self.search_beneficiary()
+        print(my_list)
+        if my_list:
+            my_list = json.loads(my_list)
+            benf_ids = [li["beneficiary"] for li in my_list]
+            print(benf_ids)
+            self.update({"duplicate_beneficiaries_ids": [(6, 0, list(benf_ids))]})
+
+    def archive_data(self):
+        beneficiary_data = self.env["openg2p.beneficiary"].browse(self.retained_id)
+
+        beneficiary_data.write({"merged_beneficiary_ids": [(4, [beneficiary_data])]})
+
+    @api.multi
+    def merge_beneficiaries(self):
+
+        idr = self.retained_id
+        print(idr)
+        beneficiary_data = self.env["openg2p.beneficiary"].browse(idr)
+        self.archive_data()
+        print(beneficiary_data)
+
+        beneficiary_data.write(
+            {
+                "first_name": str(self.firstname),
+                "last_name": str(self.firstname),
+                "email": str(self.firstname),
+                "phone": str(self.phone),
+                "street": str(self.street),
+                "street2": str(self.street),
+                "city": str(self.city),
+                "postal_code": str(self.zip),
+                "identity": str(self.identity_passport),
+                "bank": str(self.bank_account_id.bank_id.name),
+                "bank_account": str(self.bank_account_id.sanitized_acc_number),
+                "emergency_contact_name": str(self.emergency_contact),
+                "emergency_contact_phone": str(self.emergency_phone),
+            }
+        )
+
+        delete_url = "http://localhost:8080/index/" + str(idr)
```

Comment:
Move and break the URL into smalled chunk and put it in a class/global level variable. 
Deduplication_host = http://localhost:8080
Index_path = /index
search_path...
etc

Reply:
OK i will make necessary changes.